src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -71,7 +93,7 @@ internal static bool TryReadMetadata(JsonConverter converter, JsonTypeInfo jsonT
                             if ((state.Current.MetadataPropertyNames & (MetadataPropertyName.Id | MetadataPropertyName.Ref)) != 0)
                             {
                                 // No $id or $ref properties should precede $id properties.
-                                ThrowHelper.ThrowJsonException_MetadataIdIsNotFirstProperty(propertyName, ref state);
+                                ThrowHelper.ThrowJsonException_MetadataIdCannotBeCombinedWithRef(propertyName, ref state);
```

Comment:
The old message said this:

> The metadata property '$id' must be the first reference preservation property in the JSON object.

Technically this hasn't been accurate since polymorphism got introduced, since following `$id` after `$type` is supported. This narrows down the message to the actual failure mode it is still being used for.